### PR TITLE
Remove redundancy within new cuts (issue #10) 

### PR DESCRIPTION
**Pull request by frapac**

### CODE REVIEW

File: src/redund.jl [hunk 1]
```diff
@@ -38,16 +52,36 @@ function checkredundancy{T}(A::AbstractMatrix{T}, b::AbstractVector{T},
...
+
+"""Remove redundants cuts in polyhedra (A, b)."""
+function clean!{T}(A::AbstractMatrix{T}, b::AbstractVector{T},
+                   isfun::Bool, islb::Bool, tol::Float64)
+    nincumbents = size(A, 1)
+    redundants = checkredundancy(A, b, isfun, islb, tol)
+    if !isempty(redundants) && length(redundants) < nincumbents
+        tokeep = setdiff(collect(1:nincumbents), redundants)
+
+        A = A[tokeep, :]
```
Comment (blegat) [inline src/redund.jl]: This has no effect on the matrices `A` and `b` of `newcuts!`. We also need to update `mycut`.

File: src/dematos.jl [hunk 2]
```diff
@@ -37,12 +37,15 @@ type DeMatosCutPruner{N, T} <: AbstractCutPruner{N, T}
...
+    # peform exhaustive check before adding cuts
+    excheck::Bool
+
     # tolerance to check redundancy between two cuts
     TOL_EPS::Float64
 
     function DeMatosCutPruner(sense::Symbol, maxncuts::Int, tol=1e-6)
         isfun, islb = gettype(sense)
-        new(isfun, islb, spzeros(T, 0, N), T[], maxncuts, Tuple{Int64, T}[], Int[], 0, [], 0, zeros(T, 0, N), tol)
+        new(isfun, islb, spzeros(T, 0, N), T[], maxncuts, Tuple{Int64, T}[], Int[], 0, [], 0, zeros(T, 0, N), false, tol)
```
Comment (blegat) [inline src/dematos.jl]: Why is the default `false` for dematos and true of avg and decay ? Also, why don't we add it as an argument in the constructor (I would set `false` as default in the constructor).

**Review by blegat — CHANGES_REQUESTED** [review]

File: src/dematos.jl (hunk 2 above)
Comment (frapac) [reply src/dematos.jl]: That makes sense. I will modify that while rescoping this PR.

File: src/redund.jl (hunk 1 above)
Comment (frapac) [reply src/redund.jl]: Mmmmh you're right :/

**Review by blegat — APPROVED** [review]
Good to merge :)

**Pull request merged**